include/core/Discretization.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "span.hpp"
+
+using GeoDivide = int;
+
+struct Discretization
+{
+    virtual ~Discretization() = default;
+
+    enum GridType { hydrofabric_locations, spatial_structured, spatial_unstructured };
+    virtual GridType grid_type() = 0;
+
+    // Must cover an identified geographic domain
+    virtual GeoDivide divide() = 0;
+
+    virtual size_t size() = 0;
+    virtual int rank() = 0;
+
+    virtual bool equals(Discretization const* const rhs) = 0;
+};
+
+template <typename T>
+struct ConcreteDiscretization : public Discretization
+{
+    using PointType = T;
+    virtual boost::span<PointType> get_points() = 0;
+};
+
+// One point per catchment/nexus/whatever set elements
+struct HydroLocationDiscretization : public ConcreteDiscretization<int>
```

Comment:
Might need to disscuss this in more detail.  I'm tyring to understand the relationship of this struct to a `HydroLocation` as it relates to HY_Features and the conceptual model as well as the hydrofabric data model we have.

Reply:
As discussed, this is somewhat mis-named. I'll correct to follow the bit about 'points' grids above